src/Development/IDE/Plugin/CodeAction.hs

```diff
@@ -201,6 +202,21 @@ suggestRemoveRedundantImport ParsedModule{pm_parsed_source = L _  HsModule{hsmod
         = [("Remove import", [TextEdit (extendToWholeLineIfPossible contents _range) ""])]
     | otherwise = []
 
+removeAllRedundantImports :: Maybe ParsedModule -> Maybe T.Text -> [Diagnostic] -> Uri -> [CAResult]
+removeAllRedundantImports m contents diagnostics uri
+  | Just pm <- m,
+    r <- join $ map (suggestRemoveRedundantImport pm contents) diagnostics,
+    tes <- nub [ e | (_, edits) <- r, e <- edits],
```

Comment:
let's avoid nub, it's quadratic and I want ghcide to stay speedy in a module with 100 redundant imports

Reply:
It was a mistake, and I don't think `TextEdit` will duplicate, so just remove it.